examples/odfe-pipes-trace-app/client/client.py

```diff
@@ -96,65 +162,100 @@ def cleanupDB():
 
     closeCursorAndDBCnx(cursor, cnx)
 
+
 def checkout():
-    with tracer.start_as_current_span("client_checkout"):
-        checkoutAPIRequest = post(
-            "http://{}:8084/checkout".format(PAYMENT),
-            data=[
-                ("banana", 2),
-                ("orange", 3),
-                ("apple", 1),
-            ],
-        )
-        assert checkoutAPIRequest.status_code == 200
+    trace_id = None
+    try:
+        with tracer.start_as_current_span("client_checkout") as checkout_trace_group:
+            trace_id = get_hexadecimal_trace_id(checkout_trace_group.get_context().trace_id)
+            checkoutAPIRequest = post(
+                "http://{}:8084/checkout".format(PAYMENT),
+                data=[
+                    ("banana", 2),
+                    ("orange", 3),
+                    ("apple", 1),
+                ],
+            )
+            assert checkoutAPIRequest.status_code == 200
+            return get_ref_link("Checkout", "success", trace_id)
+    except:
+        return get_ref_link("Checkout", "failed", trace_id)
```

Comment:
I suggest the following if it works as expected:

```
with tracer.start_as_current_span("client_checkout") as checkout_trace_group:
    trace_id = get_hexadecimal_trace_id(checkout_trace_group.get_context().trace_id)
    try:
        checkoutAPIRequest = post(
            "http://{}:8084/checkout".format(PAYMENT),
            data=[
                ("banana", 2),
                ("orange", 3),
                ("apple", 1),
            ],
        )
        assert checkoutAPIRequest.status_code == 200
        return get_ref_link("Checkout", "success", trace_id)
    except:
        return get_ref_link("Checkout", "failed", trace_id)
```

same below.

Reply:
Actually this would work, I will probably fix it the next time when we address this.